### PR TITLE
Start implementing some secure coding methods.

### PR DESCRIPTION
**Pull request by gcasa**
Just a simple change to return an error when secured coding is asked for.   We should discuss how to actually implement secure coding.

### CODE REVIEW

File: Headers/Foundation/NSKeyedArchiver.h [hunk 1]
```diff
@@ -87,6 +87,16 @@ extern "C" {
  */
 + (NSData*) archivedDataWithRootObject: (id)anObject;
 
+  
+/**
```
Comment (fredkiefer) [inline Headers/Foundation/NSKeyedArchiver.h:91]: You should add guards for MacOS 10.13 around this method.

File: Source/NSKeyedArchiver.m [hunk 2]
```diff
@@ -460,6 +460,26 @@ + (NSData*) archivedDataWithRootObject: (id)anObject
   return AUTORELEASE(d);
 }
 
++ (NSData *) archivedDataWithRootObject: (id)anObject
```
Comment (fredkiefer) [inline Source/NSKeyedArchiver.m:463]: I think this method should not call archivedDataWithRootObject: instead it should be the other way around. This is the more general method and it should just have the code from archivedDataWithRootObject: and set the flag requiresSecureCoding on the created NSKeyedArchiver. As this flag is also still missing there is not much use in adding this method.
There is also no reason to return an error when secure coding is requested.

File: Headers/Foundation/NSKeyedArchiver.h (hunk 1 above)
Comment (gcasa) [reply Headers/Foundation/NSKeyedArchiver.h:91]: Done

File: Source/NSKeyedArchiver.m (hunk 2 above)
Comment (gcasa) [reply Source/NSKeyedArchiver.m:433]: Done

**Comment by gcasa** [comment]
I am merging this since it seems relatively simple.  This is just a stop gap until we have support.

File: Source/NSKeyedArchiver.m [hunk 3]
```diff
@@ -430,34 +430,49 @@ - (id) _encodeObject: (id)anObject conditional: (BOOL)conditional
 
 @implementation	NSKeyedArchiver
 
++ (NSData *) archivedDataWithRootObject: (id)anObject
+                  requiringSecureCoding: (BOOL)requiresSecureCoding
+                                  error: (NSError **)error
+{
+  NSData *d = nil;
+
+  if (requiresSecureCoding == NO)
```
Comment (fredkiefer) [inline Source/NSKeyedArchiver.m:439]: The idea was to call setRequiresSecureCoding: with this value on "a" further down in line 449. And of course any exceptions need to be converted in to NSError.

Comment (gcasa) [reply Source/NSKeyedArchiver.m:439]: Okay, I will open a "real" secure coding branch for this.   I will do that now.

**Pull request merged**